src/maliput_osm/osm/conversions.cc

```diff
@@ -43,8 +45,32 @@ LineString3d ToMaliput(const lanelet::ConstLineString3d& line_string) {
   return LineString3d{points};
 }
 
-Lane ToMaliput(const lanelet::Lanelet& lanelet) {
-  return {std::to_string(lanelet.id()), ToMaliput(lanelet.leftBound()), ToMaliput(lanelet.rightBound())};
+Lane ToMaliput(const lanelet::Lanelet& lanelet, const lanelet::LaneletLayer& map_layer) {
+  // Get Id.
+  const std::string id = std::to_string(lanelet.id());
+  // Get left boundary.
+  const LineString3d left_bound = ToMaliput(lanelet.leftBound());
+  // Get right boundary.
+  const LineString3d right_bound = ToMaliput(lanelet.rightBound());
+
+  // Obtains the lanelets that uses same @p bound linestring.
+  auto find_usage_of_lane_bounds = [&lanelet, &map_layer](const lanelet::ConstLineString3d& bound) {
+    const auto lanelets = map_layer.findUsages(bound);
+    MALIPUT_THROW_UNLESS(lanelets.size() >= 1);
+    MALIPUT_THROW_UNLESS(lanelets.size() <= 2);
```

Comment:
I think it makes sense to merge this into `lanelets.size() == 1 || lanelets.size() == 2`.

Did you do this to make it fail on purpose when it is 0 or more than 2 and identify which case it is?

Reply:
> Did you do this to make it fail on purpose when it is 0 or more than 2 and identify which case it is?

Exactly, there are two different situations to verify and that's why I managed them differently
1. Linestring should at least belong to one lanelet.
2. It can't be the boundary of more than two lanelets.